### DISCUSSION TITLE
Xstate for backend entity state management?

### DISCUSSION THREAD

**Discussion by sksethi25** [statelyai/xstate#4200] [Q&A]
Can we use Xstate at backend/server side for state management of an entity 

Example: Suppose to build a product like Jira where Issue entity can be moved in various status.

Sample Jira Issue flow
![jira](https://github.com/statelyai/xstate/assets/6614662/e7dfdcfc-7561-4678-9ecb-3b630d2d8bdd)

 is it feasible and efficient to achieve something like following with Xstate
Deploying Xstate Machine with Node/exrpess.js backed API that creates Issue (Suppose issue in created state) in its initial state and init the Xstate machine and then an API for giving various next Events possible (move to in progress from created state) and another API to initiate that event to make transition to next state( like moving to in progress from qa).

 Please guide further if anyone done something like this or any other apporach?

**Comment by davidkpiano** [comment]
Hey @sksethi25!

Yes, you can definitely do this. The process would look like this for a serverless function:

1. Read state from storage, if it exists.
    - If it doesn't exist, use the machine's initial state
2. Start the machine from the persisted (or initial) state
3. Send the machine the event from the `request`
4. Read the next state via `actor.getSnapshot()` and persist it to storage
5. Send a response (e.g. 200 OK)

I will make an example of this soon.

**Reply by ashubhadoria** [reply]
I am late to the party, but is there any chance, @davidkpiano, that there is some sample code for modelling this?
I have a similar use case where I need to read from a persisted storage, read the state, perform an action, and store back to that persisted storage.
